exec/java-exec/src/main/java/org/apache/drill/exec/util/JarUtil.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.util;
+
+public class JarUtil {
+
+  /**
+   * Generates source jar name based on binary jar name.
+   * It is expected binary and source have standard naming convention.
+   *
+   * @param binary binary jar name
+   * @return source jar name
+   */
+  public static String getSourceName(String binary) {
+    return binary.replace(".jar", "-sources.jar");
```

Comment:
Since for.JAR and foo.jar are two different files in Linux, I suggest we don't capitalize suffixes during check. Jars with UDFs will be generated via maven which adds .jar as lower case, so I don't expect user to change it, user should follow standard naming convention.